scipy/interpolate/regrid_python.py

```diff
@@ -669,21 +690,27 @@ def _apply_bbox_grid(x, y, Z, bbox):
 
     return x[ix], y[iy], Z[np.ix_(ix, iy)], np.s_[ix], np.s_[iy]
 
-def build_matrices(x, y, z, tx, ty, kx, ky):
+def build_matrices(x, y, z, tx, ty, kx, ky, p):
```

Comment:
Instead of adding a level of branching on `p = -1`, I'd just inline at the `s==0` branch

```
    Ax, offset_x, nc_x = _dierckx.data_matrix(x, tx, kx, np.ones_like(x))
    Ay, offset_y, nc_y = _dierckx.data_matrix(y, ty, ky, np.ones_like(y))
    Q = z.copy()
```